options/options_helper.cc

```diff
@@ -1045,21 +1045,19 @@ Status ParseColumnFamilyOption(const std::string& name,
     } else {
       if (name == kNameComparator) {
         // Try to get comparator from object registry first.
-        std::unique_ptr<const Comparator> comp_guard;
-        const Comparator* comp =
-            NewCustomObject<const Comparator>(value, &comp_guard);
         // Only support static comparator for now.
-        if (comp != nullptr && !comp_guard) {
-          new_options->comparator = comp;
+	Status status = ObjectRegistry::Get()->NewStaticObject(value, &new_options->comparator);
+        if (status.ok()) {
```

Comment:
We return when `status` is OK?

Reply:
When Status is OK, that means we successfully loaded the operator from the registry and we do not need to do further work in ParseOptionsHelper later.